lib/Sema/TypeCheckProtocol.h

```diff
@@ -210,9 +210,13 @@ enum class MatchKind : uint8_t {
   /// The witness is explicitly @nonobjc but the requirement is @objc.
   NonObjC,
 
-  /// The witness does not have a `@differentiable` attribute satisfying one
-  /// from the requirement.
-  DifferentiableConflict,
+  /// The witness is missing a `@differentiable` attribute from the requirement.
+  MissingDifferentiableAttr,
+
+  /// The witness is missing a `@differentiable` attribute from the requirement.
+  /// The witness has less-than-public visibility and is declared in a different
+  /// files than the conformance.
+  MissingDifferentiableAttrNonPublicOtherFile,
```

Comment:
Semantically`MissingDifferentiableAttr` is a superset of `MissingDifferentiableAttrNonPublicOtherFile`. What's the benefit of using two cases compared to using just one?

Reply:
Thanks for pointing this out. `MatchKind::MissingDifferentiableAttrNonPublicOtherFile` removed in https://github.com/apple/swift/pull/29771/commits/7d08e9bd8d2a7397b6643f0751530a436328485b.

---

> What's the benefit of using two cases compared to using just one?

In conformance type-checking, requirement-witness mismatches are separately (1) detected (returning a `MatchKind`) and (2) diagnosed (taking a `MatchKind`).

If logic for determining which diagnostic to emit in (1) is sufficiently complicated, it may be more desirable to add a `MatchKind` case rather than duplicating the logic in (2).

In this case, the logic for determining `MatchKind:: MissingDifferentiableAttrNonPublicOtherFile` wasn't complicated (witness is less than public and in separate file from conformance), so duplicating it in (2) makes more sense.